CHANGELOG.md

```diff
@@ -7,6 +7,71 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [0.35.0] - 2023-09-XX
+
+### Breaking changes
+
+-   Added support for FDI 1.18
+-   User type has changed across recipes and functions: recipe specific user types have been removed and replaced by a generic one that contains more information
+-   `createdNewUser` has been renamed to `createdNewRecipeUser`
+-   `createCode`, `consumeCode`, `createPasswordlessCode` and `consumePasswordlessCode` can now return status: `SIGN_IN_UP_NOT_ALLOWED`
+-   `signInAndUp` and `thirdPartySignInAndUp` can now return new status: `SIGN_IN_UP_NOT_ALLOWED`
+-   `sendPasswordResetEmail` can now return `status: "PASSWORD_RESET_NOT_ALLOWED"`
+-   `signIn` and `emailPasswordSignIn` can now return `SIGN_IN_NOT_ALLOWED`
+-   `signUp` and `emailPasswordSignUp` can now return `SIGN_UP_NOT_ALLOWED`
+-   The context param of `getRedirectionURL` gets an optional `user` prop (it's always defined if `createdNewRecipeUser` is set to true)
+-   Added new language translation keys
+
+### Migration
+
+#### New User structure
+
+We've added a generic `User` type instead of the old recipe specific ones. The mapping of old props to new in case you are not using account-linking:
+
+-   `user.id` stays `user.id`
+-   `user.email` becomes `user.emails[0]`
+-   `user.phoneNumber` becomes `user.phoneNumbers[0]`
+-   `user.thirdParty` becomes `user.thirdParty[0]`
+-   `user.timeJoined` is still `user.timeJoined`
+-   `user.tenantIds` is still `user.tenantIds`
+
+#### Checking if a new primary user was created
+
+In `getRedirectionURL`
+
+```
+EmailPassword.init({ // This looks the same for other recipes
+    // Other config options.
+    async getRedirectionURL(context) {
+        if (context.action === "SUCCESS") {
+            if (context.isNewRecipeUser && context.user.loginMethods.length === 1) {
+                // new primary user
+            } else {
+                // only a recipe user was created
+            }
+        }
+        return undefined;
+    }
+})
+```
+
+In `onHandleEvent`:
+
+```
+EmailPassword.init({ // This looks the same for other recipes
+    // Other config options.
+    onHandleEvent(context: EmailPasswordOnHandleEventContext) {
+        if (context.action === "SUCCESS") {
+            if (context.isNewRecipeUser && context.user.loginMethods.length === 1) {
+                // new primary user
+            } else {
+                // only a recipe user was created
+            }
+        }
+    },
+})
+```
+
```

Comment:
- Mention changes in functions like emailpassword.signin now also returns SIGN_IN_NOT_ALLOWED etc.
- New error messages added for these statuses. See the UI here: supertokens.com/docs/thirdpartyemailpassword/common-customizations/account-linking/automatic-account-linking#support-status-codes